### DISCUSSION TITLE
Asymmetric engine arrangement

### DISCUSSION THREAD

**Discussion by ingStudent** [JSBSim-Team/jsbsim#872] [Q&A]
Good morning everyone,

I have a question concerning the modeling of engine's lever arm when the latter are not symmetrical with respect to the CG. In the aircraft models that I found the disposition is symmetrical and they put in their xml files gains of 1 and -1,  in my case, I don't know how to transcribe that to jsbsim. If anyone of you can give me a hint?

Thank U.

**Comment by seanmcleod** [comment]
What gains are you referring to? Some PID control gains for controlling the thrust of these asymmetric engines? Give an example with more details.

**Comment by ingStudent** [comment]
<channel name="Control Mixer">
    <pure_gain name="cmdHeave_mps_2_FR">
      <input>fcs/cmdHeave_nd</input>
      <gain>1.0</gain>
      <output>fcs/cmdHeave_nd_2_FR</output>
    </pure_gain>
    <pure_gain name="cmdRoll_rps_2_FR">
      <input>fcs/cmdRoll_rps</input>
      <gain>-1.0</gain>
      <output>fcs/cmdRoll_rps_2_FR</output>
    </pure_gain>
    <pure_gain name="cmdPitch_rps_2_FR">
      <input>fcs/cmdPitch_rps</input>
      <gain>1.0</gain>
      <output>fcs/cmdPitch_rps_2_FR</output>
    </pure_gain>
    <pure_gain name="cmdYaw_rps_2_FR">
      <input>fcs/cmdYaw_rps</input>
      <gain>1.0</gain>
      <output>fcs/cmdYaw_rps_2_FR</output>
    </pure_gain>
    <summer name="cmdFR_nd">
      <input>fcs/cmdHeave_nd_2_FR</input>
      <input>fcs/cmdRoll_rps_2_FR</input>
      <input>fcs/cmdPitch_rps_2_FR</input>
      <input>fcs/cmdYaw_rps_2_FR</input>
      <output>fcs/cmdFR_nd</output>

I'm speaking about these gains. According to what it is told to me, to model this assymetrie we have to change the gains in which, in the code I gave, are 1 and -1. From what I understand, the contribution of each engine is different according to their distance from the CG.

I don't know if what I m saying is clear. 

Thank U.

**Comment by ingStudent** [comment]
```xml
<channel name="Control Mixer">
    <pure_gain name="cmdHeave_mps_2_FR">
      <input>fcs/cmdHeave_nd</input>
      <gain>1.0</gain>
      <output>fcs/cmdHeave_nd_2_FR</output>
    </pure_gain>
    <pure_gain name="cmdRoll_rps_2_FR">
      <input>fcs/cmdRoll_rps</input>
      <gain>-1.0</gain>
      <output>fcs/cmdRoll_rps_2_FR</output>
    </pure_gain>
    <pure_gain name="cmdPitch_rps_2_FR">
      <input>fcs/cmdPitch_rps</input>
      <gain>1.0</gain>
      <output>fcs/cmdPitch_rps_2_FR</output>
    </pure_gain>
    <pure_gain name="cmdYaw_rps_2_FR">
      <input>fcs/cmdYaw_rps</input>
      <gain>1.0</gain>
      <output>fcs/cmdYaw_rps_2_FR</output>
    </pure_gain>
    <summer name="cmdFR_nd">
      <input>fcs/cmdHeave_nd_2_FR</input>
      <input>fcs/cmdRoll_rps_2_FR</input>
      <input>fcs/cmdPitch_rps_2_FR</input>
      <input>fcs/cmdYaw_rps_2_FR</input>
      <output>fcs/cmdFR_nd</output>
```

**Comment by seanmcleod** [comment]
So why not just scale the gain based on the ratio of the different arm differences?

**Comment by ingStudent** [comment]
That's what I'm asking for. If we only can change the gains according to the arm or we have to meddle or do something else to the section where the engines are defined.

**Comment by seanmcleod** [comment]
We have no idea of your particular physical model, or any details on how you plan to control it, i.e. your control system so it's virtually impossible to give you a useful answer. There are lots of different ways of modelling and lots of different control options and with virtually no info to go on.

You'll need to narrow things down and give a more detailed description of what you're trying to do and where you are having an issue.

P.S
Please start sentences with a capital letter, both in the subject line and within the comments. Makes it easier to read.

**Comment by ingStudent** [comment]
Ok, I'll check and from what I'm readind, all my sentences start with a capital letter, so .... Thank U.

**Reply by seanmcleod** [reply]
I edited some of your replies, including the subject line etc. to fix up the missing capital letters.

![image](https://user-images.githubusercontent.com/824856/228494130-5c904012-6c06-4759-a7b8-3830df41d5b0.png)
